File: BackEnd/MicroServices/InternshipOffer/src/main/java/com/example/internshipoffer/Service/EncadrementService.java
package com.example.internshipoffer.Service;

import com.example.internshipoffer.Entity.Encadrement;

import java.util.List;
import java.util.Optional;

public interface EncadrementService {
    Encadrement createEncadrement(Encadrement encadrement);
    List<Encadrement> getAllEncadrements();
    Optional<Encadrement> getEncadrementById(Long id);
    Encadrement updateEncadrement(Long id, Encadrement encadrementDetails);
    void deleteEncadrement(Long id);
}

File: FrontEnd/src/app/services/encadrement.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Encadrement } from '../models/Encadrement';

@Injectable({
  providedIn: 'root'
})
export class EncadrementService {
  private apiUrl = 'http://localhost:8083/encadrements'; 

  constructor(private http: HttpClient) {}

  getEncadrements(): Observable<Encadrement[]> {
    return this.http.get<Encadrement[]>(this.apiUrl);
  }

  getEncadrementById(id: number): Observable<Encadrement> {
    return this.http.get<Encadrement>(`${this.apiUrl}/${id}`);
  }

  createEncadrement(encadrement: Encadrement): Observable<Encadrement> {
    return this.http.post<Encadrement>(this.apiUrl, encadrement);
  }

  updateEncadrement(id: number, encadrement: Encadrement): Observable<Encadrement> {
    return this.http.put<Encadrement>(`${this.apiUrl}/${id}`, encadrement);
  }

  deleteEncadrement(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
